internal/connector/test/integration/features/connector-old-path.feature

```diff
@@ -21,10 +21,6 @@ Feature: the old connectors path are still valid
           "name": "example 1",
           "kafka_id":"mykafka"
         },
-        "deployment_location": {
```

Comment:
I think we can delete this feature at all as we don't have such paths anymore (if so, we should remove them)

Reply:
Agreed, if nobody is using the old paths we should get rid of this feature test.

Reply:
@valdar do you see any issues in deleting this old path test feature?

Reply:
@dhirajsb  since the tests are passing, it means that path is served by some part of the codebase, we should get rid of that as well. I see no evident issues in doing so.

Reply:
I am getting rid of all old paths with `-` in them, so that only paths with `_` in the path prefix are supported in compliance with the latest connector openapi yaml files. 
Also, I found routes for this path in the route loader, which is also going to be removed:
```
/api/connector_mgmt/v1/kafka_connectors_of/{connector_type_id}
```